pinot-common/src/main/java/com/linkedin/pinot/common/exception/QueryException.java

```diff
@@ -88,15 +88,23 @@
   }
 
   public static ProcessingException getException(ProcessingException processingException, Exception exception,
-      int sizeOfStackTraceToTruncate) {
+      int maxLinesOfStackTrace) {
+    String errorType = processingException.getMessage();
     ProcessingException retProcessingException = processingException.deepCopy();
-    StringWriter sw = new StringWriter(sizeOfStackTraceToTruncate);
-    exception.printStackTrace(new PrintWriter(sw));
-    retProcessingException.setMessage(sw.toString());
+    StringWriter stringWriter = new StringWriter();
+    exception.printStackTrace(new PrintWriter(stringWriter));
+    String fullStackTrace = stringWriter.toString();
+    String[] lines = fullStackTrace.split("\n");
+    int numLinesOfStackTrace = Math.min(lines.length, maxLinesOfStackTrace);
+    int lengthOfStackTrace = numLinesOfStackTrace - 1;
+    for (int i = 0; i < numLinesOfStackTrace; i++) {
+      lengthOfStackTrace += lines[i].length();
+    }
+    retProcessingException.setMessage(errorType + ":\n" + fullStackTrace.substring(0, lengthOfStackTrace));
     return retProcessingException;
   }
 
   public static ProcessingException getException(ProcessingException processingException, Exception exception) {
-    return getException(processingException, exception, 1000);
+    return getException(processingException, exception, 10);
```

Comment:
Make this configurable at the server level. I feel 10 might be too less for exception stack.